connector/src/main/scala/com/datastax/spark/connector/cql/Scanner.scala

```diff
@@ -16,7 +16,10 @@ import com.datastax.spark.connector.writer.RateLimiter
   */
 trait Scanner {
   def close(): Unit
+  @deprecated("This method will be removed. The session returned by this method may not contain all the " +
+    "configuration settings", "3.1.0")
   def getSession(): CqlSession
+  def prepare(cql: String): PreparedStatement = getSession().prepare(cql)
```

Comment:
Question: Probably out of this ticket's scope, but just wondering if it may be better not to reuse the `getSession()` method here as it will be deprecated.

Reply:
I'll stick with getSession